data/libs/systems/hireling.lua

```diff
@@ -313,6 +313,7 @@ function Hireling:save()
 	sql = sql .. " WHERE `id`=" .. tostring(self.id)
 
 	db.query(sql)
+	return true
```

Comment:
Its better to return the result of `db.query` since this method returns a boolean with the query status, if it was successful or not.

db_functions.cpp
```c++
int DBFunctions::luaDatabaseExecute(lua_State* L) {
	pushBoolean(L, Database::getInstance().executeQuery(getString(L, -1)));
	return 1;
}
```

db_functions.hpp
```c++
registerMethod(L, "db", "query", DBFunctions::luaDatabaseExecute);
```

This suggestion will improve the code and be more precise in the method return.
```suggestion
	return db.query(sql)
```

Could you please modify and test it?

Reply:
Another point is that this not solves #2720, just solves the issue of the hirelings saves warning. There's no enough information to assume that the issue is related to the hirelings.